src/hdmf/container.py

```diff
@@ -132,39 +135,39 @@ def _field_config(self, arg_name, val, type_map):
             msg = "%s not found within loaded configuration." % self.namespace
             warn(msg)
             return val
-        else:
-            # check to see that the container type is in the config under the namespace
-            config_namespace = termset_config['namespaces'][self.namespace]
-            data_type = self.data_type
-
-            if data_type not in config_namespace['data_types']:
-                msg = '%s not found within the configuration for %s' % (data_type, self.namespace)
-                warn(msg)
-                return val
-            else:
-                # Get the ObjectMapper
-                obj_mapper = type_map.get_map(self)
-
-                # Get the spec for the constructor arg
-                spec = obj_mapper.get_carg_spec(arg_name)
-                if spec is None:
-                    msg = "Spec not found for %s." % arg_name
-                    warn(msg)
-                    return val
-
-                # Get spec attr name
-                mapped_attr_name = obj_mapper.get_attribute(spec)
-
-                config_data_type = config_namespace['data_types'][data_type]
-                try:
-                    config_termset_path = config_data_type[mapped_attr_name]
-                except KeyError:
-                    return val
-
-                termset_path = os.path.join(CUR_DIR, config_termset_path['termset'])
-                termset = TermSet(term_schema_path=termset_path)
-                val = TermSetWrapper(value=val, termset=termset)
-                return val
+
+        # check to see that the container type is in the config under the namespace
+        config_namespace = termset_config['namespaces'][self.namespace]
+        data_type = self.data_type
+
+        if data_type not in config_namespace['data_types']:
+            msg = '%s not found within the configuration for %s' % (data_type, self.namespace)
+            warn(msg)
+            return val
+
+        # Get the ObjectMapper
+        obj_mapper = type_map.get_map(self)
+
+        # Get the spec for the constructor arg
+        spec = obj_mapper.get_carg_spec(arg_name)
+        if spec is None:
+            msg = "Spec not found for %s." % arg_name
+            warn(msg)
+            return val
+
+        # Get spec attr name
+        mapped_attr_name = obj_mapper.get_attribute(spec)
+
+        config_data_type = config_namespace['data_types'][data_type]
+        try:
+            config_termset_path = config_data_type[mapped_attr_name]
+        except KeyError:
+            return val
+
+        termset_path = os.path.join(CUR_DIR, config_termset_path['termset'])
+        termset = TermSet(term_schema_path=termset_path)
+        val = TermSetWrapper(value=val, termset=termset)
+        return val
```

Comment:
The changes here
1. refactor the logic so that we have fewer nested if/else statements. Return breaks the if so we can dedent
2. Updates `configurator.path` to `configurator.paths`